src/Symfony/Component/DependencyInjection/Argument/LocatorArgument.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Argument;
+
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * Represents a service locator able to lazy load a given range of services.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class LocatorArgument implements ArgumentInterface
```

Comment:
If `ServiceLocator::get` is meant to return not only services, but also scalar values (ref: https://github.com/symfony/symfony/pull/21553#discussion_r99939370), is this name still relevant? Shouldn't it just be `Locator` or something similar instead? Or do we give up this to focus on services?